src/ini.pest

```diff
@@ -1,13 +1,10 @@
-WHITESPACE = _{ " " }
-char = { ASCII_ALPHANUMERIC | "." | "_" | "/" }
-name = @{ char+ }
-value = @{ char* }
+ini = { SOI ~ (section | pair | comment)* ~ EOI }
 
-section = { "[" ~ name ~ "]" }
-property = { name ~ "=" ~ value }
+section = { "[" ~ key ~ "]" }
+pair = { key ~ "=" ~ value }
+comment = { (";" | "#" ~ value }
 
-file = {
-	SOI ~
-	((section | property)? ~ NEWLINE)* ~
-	EOI
-}
+key = @{ (!("]" | "=" | WHITESPACE) ~ ANY)+ }
+value = @{ (!(NEWLINE) ~ ANY)+ }
+
+WHITESPACE = _{ " " | "\t" | NEWLINE }
```

Comment:
How do you intend to separate the newline from the rest of the whitespace? Something with slicing?

Reply:
Good question - i am not entirely sure how to handle it but probably i'll just add an at-sign to the `NEWLINE`-expression and catch it.